jenkins-agent

```diff
@@ -118,5 +118,18 @@ else
 	#TODO: Handle the case when the command-line and Environment variable contain different values.
 	#It is fine it blows up for now since it should lead to an error anyway.
 
-	exec $JAVA_BIN $JAVA_OPTS -cp /usr/share/jenkins/agent.jar hudson.remoting.jnlp.Main -headless $TUNNEL $URL $WORKDIR $WEB_SOCKET $DIRECT $PROTOCOLS $INSTANCE_IDENTITY $OPT_JENKINS_SECRET $OPT_JENKINS_AGENT_NAME "$@"
+        FUTURE_OPTS=""
+        if [ "$JENKINS_ENABLE_FUTURE_JAVA" ] ; then
```

Comment:
is this set somewhere

Reply:
my bad. I had not pushed that change. It should be there now

Reply:
also needs adding to the other java 17 image

Reply:
ye I wasn't sure to do that. I haven't tested those in the wild, don't really have scope to do that. Maybe you have tests running against the PR which cover that?

If not, but you're happy for me to add this on the basis of it being sanity checked at code level only, I can certainly do that.

Reply:
don't worry about testing that bit, we have tests covering it still runs

Reply:
done